packages/web-client/tests/acceptance/pay-test.ts

```diff
@@ -287,6 +297,59 @@ module('Acceptance | pay', function (hooks) {
     assert.dom(PAYMENT_URL).containsText(expectedUrl);
   });
 
+  test('it rounds amount up to min spend amount if currency is non-USD and non-SPEND', async function (assert) {
+    const minJpyAmount = minUsdAmount * mirageConversionRate;
+    const jpyAmount = minJpyAmount - 0.1;
+    const convertedMinSpendAmount = usdToSpend(
+      minJpyAmount / mirageConversionRate
+    );
+
+    await visit(
+      `/pay/${network}/${merchantSafe.address}?amount=${jpyAmount}&currency=${jpySymbol}`
+    );
+
+    assert
+      .dom(AMOUNT)
+      .containsText(convertAmountToNativeDisplay(minJpyAmount, jpySymbol));
+    assert.dom(SECONDARY_AMOUNT).containsText(`§${convertedMinSpendAmount}`);
+
+    let expectedUrl = generateMerchantPaymentUrl({
+      domain: universalLinkDomain,
+      network,
+      merchantSafeID: merchantSafe.address,
+      currency: jpySymbol,
+      amount: minJpyAmount,
+    });
+    assert.dom(QR_CODE).hasAttribute('data-test-styled-qr-code', expectedUrl);
+    assert.dom(PAYMENT_URL).containsText(expectedUrl);
+  });
+
+  test('it handles errors in fetching exchange rates gracefully', async function (this: MirageTestContext, assert) {
+    this.server.get('/exchange-rates', function () {
+      return new MirageResponse(502, {}, '');
+    });
+
+    const jpyAmount = 300;
+
+    await visit(
+      `/pay/${network}/${merchantSafe.address}?amount=${jpyAmount}&currency=${jpySymbol}`
+    );
+
+    assert
+      .dom(AMOUNT)
+      .containsText(convertAmountToNativeDisplay(jpyAmount, jpySymbol));
+
+    let expectedUrl = generateMerchantPaymentUrl({
+      domain: universalLinkDomain,
+      network,
+      merchantSafeID: merchantSafe.address,
+      currency: jpySymbol,
+      amount: jpyAmount,
+    });
+    assert.dom(QR_CODE).hasAttribute('data-test-styled-qr-code', expectedUrl);
+    assert.dom(PAYMENT_URL).containsText(expectedUrl);
+  });
```

Comment:
I’m glad to see this explicit handling of a failure fetching exchange rates, should there be some assertion that indicates how the UI differs in this case though? Like at least `assert.dom(SECONDARY_AMOUNT).doesNotExist()`

Reply:
I think I accidentally removed that. Thanks!